platform/pulp/platform/fields.py

```diff
@@ -0,0 +1,80 @@
+"""
+Custom django fields provided by pulp platform
+"""
+import json
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import six, ugettext_lazy as _
+
+
+class JSONField(models.TextField):
+    """
+    A custom django field to serialize dicts into a text field
+
+    """
+    def from_db_value(self, value, expression, connection, context):
+        """
+        Converts a value as returned by the database to a Python object
+
+        :param value: The value to convert to Python
+        :type value: Any
+
+        :param expression:
+        :param connection:
+        :param context:
+
+        :return: A python representation of value
+        :rtype: Any
+        """
+        if isinstance(value, six.string_types):
+            return self.to_python(value)
+
+        return value
+
+    def to_python(self, value):
+        """
+        Converts the value into the correct Python object
+
+        :param value: The value to convert to Python
+        :type value: Any
+
+        :return: A python representation of value
+        :rtype: Any
+        """
+        try:
+            return json.loads(value)
+        except Exception:
+            raise ValidationError(_("Invalid JSON stored in database"))
```

Comment:
Pulp does this pattern in a lot of places and it is difficult to debug. I think anytime we catch an exception and mutate it we should log the exception at error level. Assuming you have a logger as `_logger` you could add the line which will log the entire traceback and the bad data as well.

`_logger.exception(_('The field "value" cannot be JSON deserialized. "value" has the contents: %s') % value)`